tests/integration-tests/tests/capacity_reservations/test_capacity_blocks.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+import random
+import time
+
+import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from utils import retrieve_cfn_resources
+
+
+@pytest.mark.usefixtures("os", "scheduler")
+def test_capacity_blocks(
+    region, pcluster_config_reader, odcr_stack, clusters_factory, test_datadir, scheduler_commands_factory
+):
+    """
+    Test that clustermgtd and the system works as expected when using Capacity Blocks.
+
+    The Capacity Blocks behaviour is totally simulated, by using an ODCR and some mocking and overrides.
+
+    Fleet-config and describe-capacity-reservations output will be mocked in the following way:
+    - static nodes initially will be created because part of an ODCR
+    - fleet-config.json will be modified to consider the Capacity Reservation a CB, rather than an ODCR
```

Comment:
Since the CB is being mocked starting from an ODCR, I'm wondering if the ODCR is needed at all, and eventually mock everything without the need to have an ODCR at all.

Reply:
Removed, nice suggestion.